### DISCUSSION TITLE
optimistic updates

### DISCUSSION THREAD

**Discussion by jdgamble555** [supabase/supabase#1753] [Feature Requests]
Are you guys working on optimistic updates like firebase does... ?

Basically on any add or modify of a table, it would automatically update the DOM (or cache) asynchronously while waiting on the confirmation from the database to change...

This is how firebase works.  I know this is a similar question to offline support, but I look at this differently.  You could technically do this in react or angular on its own by setting values to a variable that binds to the DOM and gets updated by the websockets, but doing this automatically would be a plus.

Thanks,

J

**Comment by kiwicopple** [comment]
Hey @jdgamble555 - we don't have this yet but we definitely plan to do it in the future (probably as the "step before" offline). 

We don't do any state management on our client libraries yet, and when we get around to it we want to make sure it's easy and people will be willing to use it over other state management tools. Current thoughts are that it will be a reactive store, similar to Svelte or MobX's reactivity.

**Comment by madeleineostoja** [comment]
This is super exciting to hear! I know it's impossible to answer accurately but is there any (very rough) ETA for this /  is it being actively worked on?

**Comment by ummahusla** [comment]
Is this still in the plans? Been a few years since the original convo started.

**Comment by thisislvca** [comment]
Hey @kiwicopple just wanted to bump this up! It'd be a game changer feature, and would love to know if you guys still plan to add it in the future :)

**Comment by plgingras88** [comment]
I had to create a basic solution in my Supabase code to make it "optimistically resilient." It works, but it's not a long-term fix, especially in large-scale platforms. My in-house fix has dead angles for sure and simple or experimental features like experimental_useOptimistic from React are obviously not made to handle such complexity when it comes to the big picture. People needing real-time data often turn to reliable front-end libraries like TanStack’s useQuery or Apollo. Adding such a feature to Supabase, given its real-time focus, seems like a straightforward improvement. Thanks!

**Comment by s0h311** [comment]
In case anyone using react needs optimistic updates, there is the `experimental_useOptimistic` hook.
https://www.youtube.com/watch?v=M3mGY0pgFk0
